### ISSUE TITLE
socket error 97 on Raspberry Pi 2

### ISSUE THREAD

**Issue by sigwo** [PaycoinFoundation/paycoin#201]
```
...snippet removed...
Error: Couldn't open socket for incoming connections (socket returned error 97)
...snippet removed...
sending getdata: block 9274df2b4e14df5d4a7c
received block af38189419c3005de376
GetMyExternalIP() received [71.178.168.108] 71.178.168.108:0
GetMyExternalIP() returned 71.178.168.108
AddLocal(71.178.168.108:8998,4)
connection timeout
trying connection 162.207.41.21:8998 lastseen=5820.1hrs
Postponing 4 reconnects
REORGANIZE
REORGANIZE: Disconnect 0 blocks; b0b7f66a093a1d428b0d..b0b7f66a093a1d428b0d
REORGANIZE: Connect 1 blocks; b0b7f66a093a1d428b0d..693d5aadb8c72d282395
```
and then dies. I have to restart and the process above starts over. This is a fresh install with no prior blockchain.

**Comment by IngCr3at1on** [comment]
Quick research suggests this might be more related to dependencies and the distro you're running but I haven't played around with Raspberry Pis all that much...

What distro are you running on your Pi? @PaycoinMaster you still have a Pi running the paycoin wallet floating around somewhere?

**Comment by sigwo** [comment]
I have 0.3.0 running fine. It is also not wanting to sync. I guess I'll
keep restarting.

On Thursday, September 10, 2015, Nathan Bass <notifications@github.com>
wrote:

> Quick research suggests this might be more related to dependencies and the
> distro you're running but I haven't played around with Raspberry Pis all
> that much...
>
> What distro are you running on your Pi? @PaycoinMaster
> <https://github.com/PaycoinMaster> you still have a Pi running the
> paycoin wallet floating around somewhere?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PaycoinFoundation/paycoin/issues/201#issuecomment-139277705>
> .
>


-- 
Steven
http://sigwo.com

**Comment by IngCr3at1on** [comment]
To be clear... the 0.3.0 wallet isn't running at the same time is it (not that that should cause a socket error but I'm curious). Also if you want help I need you to answer my question whether you have a previous version running or not. What distro are you running on your Pi?

**Comment by sigwo** [comment]
Not running at the same time. Running raspbian. No other wallets running
atm.

On Thursday, September 10, 2015, Nathan Bass <notifications@github.com>
wrote:

> To be clear... the 0.3.0 wallet isn't running at the same time is it (not
> that that should cause a socket error but I'm curious). Also if you want
> help I need you to answer my question whether you have a previous version
> running or not. What distro are you running on your Pi?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PaycoinFoundation/paycoin/issues/201#issuecomment-139284038>
> .
>


-- 
Steven
http://sigwo.com

**Comment by IngCr3at1on** [comment]
That's very interesting... I just went through the merged pull requests (between 0.3.0 and now) to refresh my memory and there's nothing that really should effect socket handling. 

General research points to this possibly being related to IPv6 issues with specific kernel versions but that's based on like 5 minutes searching so I could be way off base.

Is it safe to assume you're using the same conf files for the individual wallets?

**Comment by sigwo** [comment]
I'm thinking it is IPv6 related. There is only one wallet right now on the test rig. I delete old ones before I start testing a new one. I'll circle back to XPY 3.0.1 later.
